lib/ex_ice/ice_agent/impl.ex

```diff
@@ -624,42 +618,24 @@ defmodule ExICE.ICEAgent.Impl do
   ## BINDING REQUEST HANDLING ##
   defp handle_binding_request(ice_agent, socket, src_ip, src_port, msg) do
     with :ok <- check_username(msg, ice_agent.local_ufrag),
-         {:ok, key} <- authenticate_msg(msg, ice_agent.local_pwd),
+         :ok <- authenticate_msg(msg, ice_agent.local_pwd),
          {:ok, prio_attr} <- get_prio_attribute(msg),
          {:ok, role_attr} <- get_role_attribute(msg),
          {:ok, use_cand_attr} <- get_use_cand_attribute(msg),
-         {{:ok, ice_agent}, _} <- {check_req_role_conflict(ice_agent, role_attr), key} do
-      case find_host_cand(ice_agent.local_cands, socket) do
-        nil ->
```

Comment:
That's no longer possible. In the previous version, after doing an ICE restart, we were deleting candidates while still maintaining sockets (to keep data stream on already found path). Now, every socket has to correspond to some local candidate.